test/libyul/StackReuseCodegen.cpp

```diff
@@ -345,6 +345,69 @@ BOOST_AUTO_TEST_CASE(reuse_slots_function_with_gaps)
 	);
 }
 
+BOOST_AUTO_TEST_CASE(reuse_on_decl_assign_to_last_used)
+{
+	string in = R"({
+		let x := 5
+		let y := x // y should reuse the stack slot of x and this should not generate any code
+		sstore(y, y)
+	})";
+	BOOST_CHECK_EQUAL(assemble(in),
+		"PUSH1 0x5 "
+		"DUP1 SWAP1 POP "
```

Comment:
Can you also add a peephole optimizer rule to remove those?

Reply:
I'll first try to finish up the memory escalator on top of this, so that we can hopefully get that to a reasonable state this month, that might get tense already - if I can't manage to get back to this soon enough, I'll create an issue for it.

Reply:
Added the rule: https://github.com/ethereum/solidity/pull/9387

Reply:
https://github.com/ethereum/solidity/issues/9386 is not an issue for this.